src/ray/raylet/scheduling/cluster_task_manager.h

```diff
@@ -128,9 +132,47 @@ class ClusterTaskManager {
   /// \param[in] any_pending True if there's any pending exemplar.
   /// \return True if any progress is any tasks are pending.
   bool AnyPendingTasks(Task *exemplar, bool *any_pending, int *num_pending_actor_creation,
-                       int *num_pending_tasks) const;
+                       int *num_pending_tasks) const override;
 
-  std::string DebugString() const;
+  /// Return the resources that were being used by this worker.
+  void FreeLocalTaskResources(std::shared_ptr<WorkerInterface> worker) override;
+
+  /// When direct call task is blocked, the worker who is executing the task should give
+  /// up the cpu resources allocated for the running task for the time being and the
+  /// worker itself should also be marked as blocked.
+  bool ReleaseCpuResourcesAndMarkWorkerAsBlocked(std::shared_ptr<WorkerInterface> worker,
```

Comment:
Marking a worker as blocked can be done in the node manager itself. Because this should be the same regardless of old or new cluster task manager.